### PR TITLE
stop backprop to the generator by detaching

### PR DESCRIPTION
**Pull request by Lan-st**
update: Sorry, I found I was wrong. Without detaching the generator won't be updated either. It seems that detaching only helps reduce the computation.
——
GAN is trained in an adversarial manner. Without detaching we will update generator when we are trying to minimize GANLoss, but generator should only be updated when we are trying to maximize the loss.

### CODE REVIEW

**Comment by batsa003** [comment]
I see. Do you know what else can go wrong with my code?
